packages/core/src/browser/test/mock-storage-service.ts

```diff
@@ -14,14 +14,24 @@ import { injectable } from 'inversify';
 @injectable()
 export class MockStorageService implements StorageService {
     readonly data = new Map<string, {} | undefined>();
+    onSetDataCallback?: (key: string, data?: any) => void;
```

Comment:
We should use event for such APIs to support multiple listeners. But in this case an event seems to be redundant: https://github.com/theia-ide/theia/pull/2093/files/c116ae8ac34a36fec80c66a734982661ed3ff3cb#r196008736